torch/_inductor/ir.py

```diff
@@ -1310,7 +1313,7 @@ def create_multilayer_existing_ranges(
             wrapper_fn,
             original_ranges,
             original_reduction_ranges,
-            new_ranges,
+            [*original_ranges, *new_ranges],
```

Comment:
I feel this is a bit confusing. If `keep_dim=True`, isn't it the case that `new_ranges` already includes `original_ranges`? (Haven't touched this code for a while, I may not remember it well. It might help with review if you could help paste generated ir nodes.)

Reply:
I'm also puzzled, but I had to add it because of this check: https://github.com/pytorch/pytorch/blob/3de78a1b48d2bfdf2dba89c983df518f2120fecf/torch/_inductor/ir.py#L1234

Reply:
And I believe that at this stage, `new_ranges` is just the ranges of the preceding node, thus without the leading ones.